PatternPal/PatternPal.LoggingServer/Protos/logcollector.proto

```diff
@@ -0,0 +1,113 @@
+﻿syntax = "proto3";
+
+option csharp_namespace = "PatternPal.LoggingServer";
+
+package log;
+
+// The greeting service definition.
+
+service LogCollectorService {
+  // Sends a greeting
+  rpc Log (LogRequest) returns (LogResponse);
+}
+
+// Input arguments for recognizer request.
+message LogRequest {
+
+// Specific columns 
+EventType event_type = 1;
+string event_id = 2;
+string subject_id = 3;
+string tool_instances = 4;
+string code_state_id = 5;
+
+// Optional columns
+string client_timestamp = 6; // ISO 8601
+string experimental_condition = 7; 
+string logging_error_id = 8;
+string parent_event_id = 9;
+string session_id = 10;
+string project_id = 11;
+string code_state_section = 12;
+EventInitiator event_initiatior = 13;
+EditType edit_type = 14;
+string compile_result = 15;
+string compile_message_type = 16;
+string compile_message_data = 17;
+string source_location = 18;
+string execution_id = 19;
+string test_id = 20;
+string execution_result_id = 21;
+string intervention_category = 22;
+string intervention_type = 23;
+string intervention_message = 24;
+bytes data = 25;
+
+}
+
+// Input arguments for recognizer response.
+message LogResponse {
+
+    string message = 1;
+
+}
+
+// The possible event types that are logged
+
+enum EventType
+{
+    ET_UNKNOWN = 0;
+    ET_SESSION_START = 1;
+    ET_SESSION_END = 2;
+    ET_PROJECT_OPEN = 3;
+    ET_PROJECT_CLOSE = 4;
+    ET_FILE_CREATE = 5;
+    ET_FILE_DELETE = 6;
+    ET_FILE_OPEN = 7;
+    ET_FILE_CLOSE = 8;
+    ET_FILE_FOCUS = 9;
+    ET_FILE_RENAME = 10;
+    ET_FILE_COPY = 11;
+    ET_FILE_EDIT = 12;
+    ET_COMPILE = 13;
+    ET_COMPILE_ERROR = 14;
+    ET_COMPILE_WARNING = 15;
+    ET_SUBMIT = 16;
+    ET_RUN_PROGRAM = 17;
+    ET_DEBUG_PROGRAM = 18;
+    ET_RUN_TEST = 19;
+    ET_DEBUG_TEST = 20;
+    ET_RESOURCE_VIEW = 21;
+    ET_INTERVENTION = 22;
+    ET_CUSTOM = 23;
+}
+
+// The possible event initiators
+enum EventInitiator
+{
+    USER_DIRECT_ACTION = 0;
```

Comment:
Enum values in Protocol Buffers should have a prefix, as they're in the global scope, see the [documentation](https://protobuf.dev/programming-guides/proto3/#enum).

The `0` value also shouldn't be used.